### ISSUE TITLE
Bug found in repos_upload()

### ISSUE THREAD

**Issue by lagrelax** [rstudio/packrat#264]
Hi,

I've resolved my issue in [Issue 257](https://github.com/rstudio/packrat/issues/257). When calling `repos_upload('myLib','local_cran')`, packrat puts the value of `'local_cran'` in my repos to the `Repository` field of the `DESCRIPTION` file of myLib. For instance, if my `repos` is like:

  `                                    CRAN                                  CRANextra `
`               "https://cran.rstudio.com/"       "http://www.stats.ox.ac.uk/pub/RWin" `
`                                     local_cran `
`"file:///D:/local_cran" `

Then the `DESCRIPTION` file in the myLib in the local_cran is something like:
`...`
`Repository: file:///D:/local_cran`
`...`

The related code is  

`repo <- if (isNameOfRepo) repos[[to]] else to`

in line 113 of `cranlike-repositories.R`

However, in the function `inferPackageRecord` (line 299 of `pkg.R`):
`else if (!is.null(df$Repository) && df$Repository %in% names(repos)) `
`{`
`    # It's a package from a custom CRAN-like repo!`
`...`

It only checks if the value of `Repository` is in the names of `repos`, so what it actually wants is `Repository: local_cran`

Now I have to manually change my DESCRIPTION file to go through the `snapshot()`. Thanks.

**Comment by kevinushey** [comment]
Thanks for the investigation! Indeed we're writing out the wrong 'Repository:' field here -- I'll try to make a fix soon.

**Comment by lagrelax** [comment]
Thanks. Also seems packrat only likes `repos` as the name of the local_cran location.

**Comment by kevinushey** [comment]
The following sequence of commands correctly tags the `Repository:` field for me with the latest version of `packrat` (using it to upload and install `packrat` itself):

```R
name <- "local-cran"
path <- file.path(tempdir(), name)
packrat::repos_create(path)
packrat::repos_upload("packrat", name)
install.packages("packrat", repos = getOption("repos")[[name]])
```

Can you please post the code that's giving you trouble?

**Comment by lagrelax** [comment]
No. Still have the full path in the `Repository` field. My packrat version is `0.4.6-1`.

```
name<-"local_cran"
path<-file.path('D:/Project',name)
packrat::repos_create(path)
packrat::repos_upload('myPack',name)
```
`Repository: file:///D:/Project/local_cran`

**Comment by kevinushey** [comment]
Can you please install the development version with `devtools::install_github('rstudio/packrat')` and try again?
